### DISCUSSION TITLE
[hardware] display device for inscriptions

### DISCUSSION THREAD

**Discussion by punk3700** [ordinals/ord#1547] [Ideas]
gm. we're designing and manufacturing a display device for generative art.  currently, we're supporting ethereum (artblocks, etc) and tezos (fxhash).

curious to know if there is a demand for displaying inscriptions?

our team would be open to building integration with ordinals.  the display will fetch the content of the inscription and display it.

here are some references:

https://twitter.com/punk3700/status/1612407982029164546

https://twitter.com/punk3700/status/1612776814749061120

**Comment by casey** [comment]
This would be sick! Let me know if API support is required. An API (see #1249) is on the roadmap, but we need an initial consumer of the API to make sure we build something useful that meets a real need, and something that would display inscriptions would be a great first use-case!

**Comment by punk3700** [comment]
happy to be the first developer to consume the API.  we need a couple of features:

1) fetch an inscription by a specific id. in our mobile app, people will search for an id, and then "stream" it onto the display device.

2) fetch all inscriptions (probably paging?). we also have a browse tab in the mobile app, where people browse through different artwork and pick one they like to display.

when will the API be available?

**Reply by casey** [reply]
The best way to get started is to pick a single endpoint, for example, fetching an inscription by ID, and then opening an issue for that. I want to add things one at a time. If you opened an issue for fetching an inscription by ID, that could be added easily.

The API would be un-authed, because no private data, and you would have to run your own `ord` instance, but if that works, just adding individual endpoints is very easy.
